homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,207 @@
+) -> None:
+    """Set up the sensor platform."""
+    _LOGGER.debug("setup sensor for platform")
+    session = async_get_clientsession(hass)
+
+    if CONF_BASEURL in config:
+        _LOGGER.warning("Baseurl is not supported anymore.")
+
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    districts = config[CONF_DISTRICTS]
+
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district[CONF_DISTRICT_NAME], info_type)
+        for info_type in SENSORS
+        for district in districts
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Create sensors from a config entry in the integrations UI."""
+    _LOGGER.debug(f"create sensor from config entry {config_entry.data}")
+    session = async_get_clientsession(hass)
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    try:
+        district = config_entry.data[ATTR_COUNTY]
+        sensors = [
+            RKICovidNumbersSensor(coordinator, district, info_type)
+            for info_type in SENSORS
+        ]
+        async_add_entities(sensors, update_before_add=True)
+    except KeyError:
+        _LOGGER.error("Could not determine %s from config!", ATTR_COUNTY)
+
+
+class RKICovidNumbersSensor(CoordinatorEntity):
+    """Representation of a sensor."""
+
+    name = None
+    unique_id = None
+
+    def __init__(
+        self,
+        coordinator: update_coordinator.DataUpdateCoordinator,
+        district: Dict[str, str],
+        info_type: str,
+    ):
+        """Initialize a new sensor."""
+        _LOGGER.debug(f"initialize {info_type} sensor for {district}")
+        super().__init__(coordinator)
+
+        data = coordinator.data[district]
+
+        if data.county:
+            self.name = f"{data.county} {info_type}"
+        else:
+            self.name = f"{data.name} {info_type}"
+        self.unique_id = f"{district}-{info_type}"
+        self.district = district
+        self.info_type = info_type
+        self.updated = datetime.now()
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return (
+            self.coordinator.last_update_success
+            and self.district in self.coordinator.data
+        )
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return current state."""
+        try:
+            return getattr(self.coordinator.data[self.district], self.info_type)
+        except AttributeError:
+            return None
```

Comment:
Don't use state, use native value

Reply:
Where can I find details in the docs?

Reply:
https://github.com/home-assistant/core/blob/dev/homeassistant/components/sensor/__init__.py#L260

Reply:
It is recommended in the official sample and necessary to publish the sensor data
https://github.com/home-assistant/example-custom-config/blob/master/custom_components/example_sensor/sensor.py#L34

Reply:
Okey, so those examples needs to be fixed. If you see my link it implies not to use state, same with unit_of_measure.

Reply:
done

Reply:
The example on your link above is now fixed according to the new guidelines which might help.

Reply:
done